packages/prisma/migrations/20250617123959_update_handle_routing_form_response_fields/migration.sql

```diff
@@ -0,0 +1,143 @@
+                IF response_field->>'value' IS NOT NULL AND jsonb_typeof(response_field->'value') = 'array' THEN
+                    BEGIN
+                        INSERT INTO "RoutingFormResponseField" ("responseId", "fieldId", "valueStringArray")
+                        VALUES (
+                            NEW.id,
+                            field_record->>'id',
+                            ARRAY(SELECT jsonb_array_elements_text(response_field->'value'))
+                        );
+                    EXCEPTION WHEN OTHERS THEN
+                        RAISE WARNING 'Failed to insert multiselect values for field %', field_record->>'id';
+                    END;
+                END IF;
+            ELSIF field_type = 'number' THEN
+                -- Handle number values
+                -- Only insert if value is not null and is a valid number
+                IF response_field->>'value' IS NOT NULL AND jsonb_typeof(response_field->'value') = 'number' THEN
+                    BEGIN
+                        INSERT INTO "RoutingFormResponseField" ("responseId", "fieldId", "valueNumber")
+                        VALUES (
+                            NEW.id,
+                            field_record->>'id',
+                            (response_field->>'value')::decimal
+                        );
+                    EXCEPTION WHEN OTHERS THEN
+                        RAISE WARNING 'Failed to insert number value for field %', field_record->>'id';
+                    END;
+                END IF;
+            ELSIF field_type = 'select' THEN
+                -- Handle select values - can be either string or array
+                IF response_field->>'value' IS NOT NULL THEN
+                    BEGIN
+                        IF jsonb_typeof(response_field->'value') = 'array' THEN
+                            -- If it's an array, take the first element
+                            INSERT INTO "RoutingFormResponseField" ("responseId", "fieldId", "valueString")
+                            VALUES (
+                                NEW.id,
+                                field_record->>'id',
+                                (response_field->'value'->0)::text
+                            );
+                        ELSIF jsonb_typeof(response_field->'value') = 'string' THEN
+                            -- If it's a string, use it directly
+                            INSERT INTO "RoutingFormResponseField" ("responseId", "fieldId", "valueString")
+                            VALUES (
+                                NEW.id,
+                                field_record->>'id',
+                                response_field->>'value'
+                            );
+                        END IF;
+                    EXCEPTION WHEN OTHERS THEN
+                        RAISE WARNING 'Failed to insert select value for field %', field_record->>'id';
+                    END;
+                END IF;
```

Comment:
The line 90 ~ 114 is the addition to this function in this PR. The rest is the same as before (`packages/prisma/migrations/20250502114245_setup_triggers_for_routing_form_response_field/migration.sql`)